homeassistant/config_entries.py

```diff
@@ -2761,11 +2761,29 @@ def async_update_reload_and_abort(
         unique_id: str | None | UndefinedType = UNDEFINED,
         title: str | UndefinedType = UNDEFINED,
         data: Mapping[str, Any] | UndefinedType = UNDEFINED,
+        data_updates: dict[str, Any] | UndefinedType = UNDEFINED,
         options: Mapping[str, Any] | UndefinedType = UNDEFINED,
         reason: str | UndefinedType = UNDEFINED,
         reload_even_if_entry_is_unchanged: bool = True,
     ) -> ConfigFlowResult:
-        """Update config entry, reload config entry and finish config flow."""
+        """Update config entry, reload config entry and finish config flow.
+
+        :param data: used to replace the entry data with new data
+        :param data_updates: used to merge of updated data with pre-existing entry data
```

Comment:
Maybe this makes it clear data_updates wins if keys are in both dictionaries?
```suggestion
        :param data_updates: add items from data_updates to entry data
```